### ISSUE TITLE
[Bug]Crash when sharing to Fenix

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#4486]
## Prerequisites
Have a couple of open tabs
Have some history items
Have some saved bookmarks

## Steps to reproduce
1. From the open tabs section tap the 3dot button and select "Share tabs" and afterwards select "Firefox Preview"
2. Navigate to "Your library" and select "Bookmarks", tap the 3dot button nest tot the bookmark, select "Share" and afterwards afterwards select "Firefox Preview"
3. Navigate to "Your library" and select "History", long tap on any history item, select share from the upper control bar afterwards select "Firefox Preview"

### Expected behavior
The selected items are opened in a new tab in Fenix

### Actual behavior
Fenix crashes

### Device information

* Android device: 
• Motorola Moto G6 (Android  8)
• Huawei Mate 20 Lite (Android 8.1.0)
• Pixel 3a (Android 9)

* Fenix version: 
• RC 1.2-rc2
• Nightly Build #12140610 from 2/8

### Open Tabs
 
  ```
  --------- beginning of crash
2019-08-02 12:35:44.758 3009-3009/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix, PID: 3009
    java.lang.RuntimeException: Failed to call observer method
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:6)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2)
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:46)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:5)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:2)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:7)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:14)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:18)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391)
        at android.app.Activity.performStart(Activity.java:7165)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalStateException: Display already acquired
        at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:5)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:15)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:3)
        at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:9)
        at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:1)
        at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:6)
        at mozilla.components.support.base.feature.ViewBoundFeatureWrapper.start$support_base_release(ViewBoundFeatureWrapper.kt:1)
        at mozilla.components.support.base.feature.LifecycleBinding.start(ViewBoundFeatureWrapper.kt:1)
        at java.lang.reflect.Method.invoke(Native Method)
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:4)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2) 
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2) 
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3) 
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:46) 
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:5) 
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:2) 
        at androidx.fragment.app.Fragment.performStart(Fragment.java:7) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106) 
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2) 
        at androidx.fragment.app.Fragment.performStart(Fragment.java:14) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106) 
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2) 
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:18) 
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1) 
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391) 
        at android.app.Activity.performStart(Activity.java:7165) 
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975) 
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180) 
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
2019-08-02 12:35:44.763 3009-3009/? I/mozac/CrashReporter: Received crash: UncaughtExceptionCrash(throwable=java.lang.RuntimeException: Failed to call observer method)
2019-08-02 12:35:44.763 3009-3009/? I/mozac/CrashReporter: Showing prompt
2019-08-02 12:35:44.767 1181-4611/? I/ActivityManager: START u0 {flg=0x10018000 cmp=org.mozilla.fenix/mozilla.components.lib.crash.prompt.CrashReporterActivity (has extras)} from uid 10192
2019-08-02 12:35:44.770 766-766/? D/android.hardware.power@1.3-service.bonito-libperfmgr: LAUNCH ON
2019-08-02 12:35:44.775 3009-3009/? I/Process: Sending signal. PID: 3009 SIG: 9
2019-08-02 12:35:44.779 1181-1202/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-08-02 12:35:44.779 1181-1202/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-08-02 12:35:44.821 1181-1380/? W/InputDispatcher: channel '9443277 org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2019-08-02 12:35:44.821 1181-1380/? E/InputDispatcher: channel '9443277 org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-08-02 12:35:44.822 1181-1380/? W/InputDispatcher: channel '57a9e org.mozilla.fenix/org.mozilla.fenix.HomeActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2019-08-02 12:35:44.822 1181-1380/? E/InputDispatcher: channel '57a9e org.mozilla.fenix/org.mozilla.fenix.HomeActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-08-02 12:35:44.822 3075-3107/? I/Gecko: [GFX1-]: Receive IPC close with reason=AbnormalShutdown
2019-08-02 12:35:44.822 3075-3412/? I/Gecko: Exiting due to channel error.
2019-08-02 12:35:44.833 1181-1197/? I/ActivityManager: Process org.mozilla.fenix (pid 3009) has died: fore TOP 
2019-08-02 12:35:44.833 1181-1205/? W/libprocessgroup: kill(-3009, 9) failed: No such process

```
### Bookmarks

```
2019-08-02 12:37:03.710 3985-3985/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix, PID: 3985
    java.lang.RuntimeException: Failed to call observer method
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:6)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2)
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:46)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:5)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:2)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:7)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:14)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:18)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391)
        at android.app.Activity.performStart(Activity.java:7165)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalStateException: Display already acquired
        at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:5)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:15)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:3)
        at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:9)
        at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:1)
        at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:6)
        at mozilla.components.support.base.feature.ViewBoundFeatureWrapper.start$support_base_release(ViewBoundFeatureWrapper.kt:1)
        at mozilla.components.support.base.feature.LifecycleBinding.start(ViewBoundFeatureWrapper.kt:1)
        at java.lang.reflect.Method.invoke(Native Method)
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:4)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2) 
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2) 
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3) 
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:46) 
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:5) 
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:2) 
        at androidx.fragment.app.Fragment.performStart(Fragment.java:7) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106) 
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2) 
        at androidx.fragment.app.Fragment.performStart(Fragment.java:14) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106) 
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2) 
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:18) 
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1) 
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391) 
        at android.app.Activity.performStart(Activity.java:7165) 
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975) 
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180) 
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
2019-08-02 12:37:03.710 3985-4010/? W/VideoCapabilities: Unrecognized profile 4 for video/hevc
2019-08-02 12:37:03.716 3985-3985/? I/mozac/CrashReporter: Received crash: UncaughtExceptionCrash(throwable=java.lang.RuntimeException: Failed to call observer method)
2019-08-02 12:37:03.716 3985-3985/? I/mozac/CrashReporter: Showing prompt
2019-08-02 12:37:03.721 1181-1529/? I/ActivityManager: START u0 {flg=0x10018000 cmp=org.mozilla.fenix/mozilla.components.lib.crash.prompt.CrashReporterActivity (has extras)} from uid 10192
2019-08-02 12:37:03.724 766-766/? D/android.hardware.power@1.3-service.bonito-libperfmgr: LAUNCH ON
2019-08-02 12:37:03.731 3985-4010/? W/VideoCapabilities: Unrecognized profile 4 for video/hevc
2019-08-02 12:37:03.731 3985-3985/? I/Process: Sending signal. PID: 3985 SIG: 9
2019-08-02 12:37:03.735 1181-1202/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-08-02 12:37:03.735 1181-1202/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-08-02 12:37:03.774 1181-1380/? W/InputDispatcher: channel '77c791a org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2019-08-02 12:37:03.774 1181-1380/? E/InputDispatcher: channel '77c791a org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-08-02 12:37:03.782 1181-1380/? W/InputDispatcher: channel '1d51064 org.mozilla.fenix/org.mozilla.fenix.HomeActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2019-08-02 12:37:03.782 1181-1380/? E/InputDispatcher: channel '1d51064 org.mozilla.fenix/org.mozilla.fenix.HomeActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-08-02 12:37:03.783 1181-7949/? I/WindowManager: WIN DEATH: Window{77c791a u0 org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity}

```

### History

```
2019-08-02 12:38:22.405 4230-4230/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix, PID: 4230
    java.lang.RuntimeException: Failed to call observer method
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:6)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2)
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:46)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:5)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:2)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:7)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:14)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:18)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391)
        at android.app.Activity.performStart(Activity.java:7165)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalStateException: Display already acquired
        at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:5)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:15)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:3)
        at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:9)
        at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:1)
        at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:6)
        at mozilla.components.support.base.feature.ViewBoundFeatureWrapper.start$support_base_release(ViewBoundFeatureWrapper.kt:1)
        at mozilla.components.support.base.feature.LifecycleBinding.start(ViewBoundFeatureWrapper.kt:1)
        at java.lang.reflect.Method.invoke(Native Method)
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:4)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2) 
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2) 
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3) 
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:46) 
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:5) 
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:2) 
        at androidx.fragment.app.Fragment.performStart(Fragment.java:7) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106) 
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2) 
        at androidx.fragment.app.Fragment.performStart(Fragment.java:14) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:106) 
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:211) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2) 
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:18) 
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1) 
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391) 
        at android.app.Activity.performStart(Activity.java:7165) 
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2975) 
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180) 
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
2019-08-02 12:38:22.422 4230-4230/? I/mozac/CrashReporter: Received crash: UncaughtExceptionCrash(throwable=java.lang.RuntimeException: Failed to call observer method)
2019-08-02 12:38:22.423 4230-4230/? I/mozac/CrashReporter: Showing prompt
2019-08-02 12:38:22.428 1181-1796/? I/ActivityManager: START u0 {flg=0x10018000 cmp=org.mozilla.fenix/mozilla.components.lib.crash.prompt.CrashReporterActivity (has extras)} from uid 10192
2019-08-02 12:38:22.432 766-766/? D/android.hardware.power@1.3-service.bonito-libperfmgr: LAUNCH ON
2019-08-02 12:38:22.442 4230-4230/? I/Process: Sending signal. PID: 4230 SIG: 9
2019-08-02 12:38:22.445 1181-1202/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-08-02 12:38:22.445 1181-1202/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-08-02 12:38:22.490 4275-4461/? I/Gecko: [Child 4275, Unnamed thread cd213200] WARNING: pipe error (42): Connection reset by peer: file /builds/worker/workspace/build/src/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 358
2019-08-02 12:38:22.490 4275-4461/? I/Gecko: Exiting due to channel error.
2019-08-02 12:38:22.490 1181-1380/? W/InputDispatcher: channel '5445d66 org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2019-08-02 12:38:22.490 1181-1380/? E/InputDispatcher: channel '5445d66 org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-08-02 12:38:22.497 1181-1380/? W/InputDispatcher: channel '32cd880 org.mozilla.fenix/org.mozilla.fenix.HomeActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2019-08-02 12:38:22.497 1181-1380/? E/InputDispatcher: channel '32cd880 org.mozilla.fenix/org.mozilla.fenix.HomeActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-08-02 12:38:22.498 1181-5046/? I/WindowManager: WIN DEATH: Window{5445d66 u0 org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity}
```

**Comment by csadilek** [comment]
Thanks! That's different STR but same crash as in https://github.com/mozilla-mobile/fenix/issues/4438 and also https://sentry.prod.mozaws.net/operations/fenix-nightly/issues/5917855/ we've been having for 2 months +.

**Comment by sv-ohorvath** [comment]
@csadilek Should we close this as a dupe then?

**Comment by csadilek** [comment]
@sv-ohorvath good question. I think I'd like to keep it as the STRs are different and this way we can verify separately?

**Comment by csadilek** [comment]
See here for details on the cause/regression: https://github.com/mozilla-mobile/fenix/issues/4438#issuecomment-517786739

**Comment by csadilek** [comment]
This should be fixed on master and latest Nightly.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on **Fenix Nightly Build #12210609 from 9/8** using the following devices:
• Google Pixel 3a (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7)
